### ISSUE TITLE
Pftool writes fail into the root of a MarFS repository -

### ISSUE THREAD

**Issue by thewacokid** [pftool/pftool#73]
I need to replicate this on the testbed, but here's the behavior in production:

Repo is /examplerepo within the config.  Root of the filesystem is /campaign.

So:
/campaign/examplerepo

Is the root of the user-visible space.  A recursive pfcp *into* that directory from any source directory fails with:
"manager: failed to create temp_path /campaign"
and
"ERROR FATAL: parent doesn't exists: /campaign/examplerepo: /path/to/source/dir/parent"

The pfcp line would be something similar to:
pfcp -R /path/to/source/dir/parent/child /campaign/examplerepo

My quick bypass is to write to subdirectories, but this is visible to users.

**Comment by jti-lanl** [comment]
I’ll dig into this with you, but, if I understand correctly, you are trying
to create the namespace root-dir with a simple ‘mkdir’ operation, which
I believe is not allowed.  /campaign/namespace  represents an entire
namespace, and creating the dir is something an admin needs to do,
as part of the process of creating a new namespace; it involves more
than just creating a directory.

We could consider supporting something like this.  I recall there were
questions about directory ownership, etc.

**Comment by garygrider** [comment]
Letting users create their own name spaces,    run away run away

**Comment by thewacokid** [comment]
No, no, not the root of the MarFS namespace, the root of a user repository that we've already set up.

Pftool jobs into /campaign/namespace/ fail.  Mkdir through FUSE is fine within that directory.

So I set up for a user a namespace called "namespace":
/campaign/namespace/

They can't pfcp directly into that space, they have to create a directory within and then copy into that.

**Comment by jti-lanl** [comment]
Okay.

Using my development pftool code (rdma_issue_200 branch, includes recently-merged master), running as a regular user on a RHEL7 FTA, I don’t see this behavior with raw pftool.

$ mpirun -x MARFSCONFIGRC --map-by node -np 4 -H localhost pftool -r -p /dev/shm/packed -c /campaign.jti/mc -w 0
… appears to work correctly.

$ mkdir /dev/shm/dir
$ echo "small test" > /dev/shm/dir/small_test.txt
$ mpirun -x MARFSCONFIGRC --map-by node -np 4 -H localhost pftool -r -p /dev/shm/dir -c /campaign.jti/mc -w 0 
$ cat /campaign.jti/mc/dir/small_test.txt 
small test


If I set up the the namespace-dir itself to forbid write-access by my user, then I do have problems.

There is currently a problem running pfcp as user on the testbed, just to do with MPI module-loading in .bashrc, so I didn’t test that.

**Comment by thewacokid** [comment]
I'll try to replicate on the testbed - perhaps we fixed this between the release running in production and the current master.

**Comment by thewacokid** [comment]
This was fixed a long time ago.
